### PR TITLE
Update Static Variables

### PR DESCRIPTION
**Pull request by RoseSecurity**
## Why:

- Components had statically assigned values that could benefit from dynamic assignment via variables

## What:

- Updated `elasticsearch`, `bastion`, `alb-controller`, `eks`, `ecs-service`, and `snowflake` components with variable input

### CODE REVIEW

**Comment by milldr** [comment]
This looks good, but[ the "bats" tests are failing](https://github.com/cloudposse/terraform-aws-components/actions/runs/5658413325/job/15329677726?pr=777)

**Comment by milldr** [comment]
Can you add simple descriptions to each variable in `modules/eks/eks-without-spotinst/modules/node_group_by_az/variables.tf`? That's not part of this change, but it's a requirement for bats to pass

**Comment by RoseSecurity** [comment]
Of course! I'll add it shortly! Thanks for the heads up

**Comment by RoseSecurity** [comment]
Just added those descriptions and bats passed!

**Comment by RoseSecurity** [comment]
I can do that! Thank you for checking that out.

File: modules/ecs-service/variables.tf [hunk 1]
```diff
@@ -182,6 +182,12 @@ variable "stickiness_cookie_duration" {
   description = "The time period, in seconds, during which requests from a client should be routed to the same target. After this time period expires, the load balancer-generated cookie is considered stale. The range is 1 second to 1 week (604800 seconds). The default value is 1 day (86400 seconds)"
 }
 
+variable "trace_enabled" {
+  type        = string
+  default     = "true"
+  description = "Enable sending of spans to the agent"
+}
+
```
Comment (Nuru) [inline modules/ecs-service/variables.tf:190]: See above about `DD_TRACING_ENABLED`
```suggestion
```

File: modules/eks/eks-without-spotinst/modules/node_group_by_az/main.tf [hunk 2]
```diff
@@ -58,9 +58,9 @@ module "eks_node_group" {
   subnet_ids                 = local.enabled ? local.subnet_ids : null
 
   block_device_mappings = local.enabled ? [{
-    device_name           = "/dev/xvda"
+    device_name           = var.block_device_name
     volume_size           = var.cluster_context.disk_size
-    volume_type           = "gp2"
+    volume_type           = var.block_device_volume_type
```
Comment (Nuru) [inline modules/eks/eks-without-spotinst/modules/node_group_by_az/main.tf:63]: This module is deprecated, let's not make changes.

File: modules/snowflake-account/variables.tf [hunk 3]
```diff
@@ -31,6 +31,12 @@ variable "snowflake_admin_username" {
   description = "Snowflake admin username created with the initial account subscription."
 }
 
+variable "snowflake_role_name" {
+  type        = string
+  default     = "ACCOUNTADMIN"
+  description = "The name of the role being granted"
+}
+
```
Comment (Nuru) [inline modules/snowflake-account/variables.tf:39]: ACCOUNTADMIN is a hardcoded role in Snowflake. It is not configurable and therefore does not need an input.

```suggestion
```

**Review by Nuru — CHANGES_REQUESTED** [review]
`eks/eks-without-spotinst` is deprecated, let's not make any changes to it. See other comments inline.

File: modules/snowflake-account/variables.tf (hunk 3 above)
Comment (RoseSecurity) [reply modules/snowflake-account/variables.tf]: Just reverted this; thank you

File: modules/ecs-service/variables.tf (hunk 1 above)
Comment (RoseSecurity) [reply modules/ecs-service/variables.tf]: Reverted to original

**Comment by RoseSecurity** [comment]
Thank you @aknysh! Just applied those changes and formatted everything

**Comment by RoseSecurity** [comment]
Should I close this PR?

**Comment by arcaven** [comment]
I think this may have stalled. Do you want us to review/update this PR, or close this one and open a new one, or abandon these updates (they were part of batch changes targeted across many components, not something we needed per se)

**Pull request closed without merging**